From: John Crispin <john@phrozen.org>
Subject: ath79: Register GPIO driver earlier

HACK: register the GPIO driver earlier to ensure that gpio_request calls
from mach files succeed.

Submitted-by: John Crispin <john@phrozen.org>
---
 drivers/gpio/gpio-ath79.c                     | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

--- a/drivers/gpio/gpio-ath79.c
+++ b/drivers/gpio/gpio-ath79.c
@@ -300,7 +300,11 @@ static struct platform_driver ath79_gpio
 	.probe = ath79_gpio_probe,
 };
 
-module_platform_driver(ath79_gpio_driver);
+static int __init ath79_gpio_init(void)
+{
+	return platform_driver_register(&ath79_gpio_driver);
+}
+postcore_initcall(ath79_gpio_init);
 
 MODULE_DESCRIPTION("Atheros AR71XX/AR724X/AR913X GPIO API support");
 MODULE_LICENSE("GPL v2");
